engine/src/main/java/org/pentaho/di/job/entries/ftpsget/FTPSConnection.java

```diff
@@ -169,6 +173,11 @@ public void connect() throws KettleException {
       connection =
         FTPConnectionFactory.getInstance( getProperties(
           hostName, portNumber, userName, passWord, connectionType, timeOut, passiveMode ) );
+      if ( connection.getConnectionType() == FTPConnection.IMPLICIT_SSL_WITH_CRYPTED_DATA_FTP_CONNECTION
```

Comment:
Is there any resources that the previous connection object could release?

Reply:
not before connect()